omod/src/main/webapp/pages/forms/extension.gsp

```diff
@@ -55,7 +55,7 @@ ${ui.message("formentryapp." + extensionForm.uiLocation)}
 <input type="text" id="extensionForm.label" name="extensionForm.label" value="${ui.encodeHtmlContent(extensionForm.label)}"/></p>
 
 <p><label name="extensionForm.icon">${ui.message("formentryapp.icon")}</label>
-<input type="text" id="extensionForm.icon" name="extensionForm.icon" value="${extensionForm.icon}"/></p>
+<input type="text" id="extensionForm.icon" name="extensionForm.icon" value="${ui.encodeHtmlContent(extensionForm.icon)}"/></p>
```

Comment:
Do you mind throwing some light as to why you used encodeHtmlContent here and escapeAttribute for the others?

Reply:
the icon field seems to have been missed in this commit https://github.com/openmrs/openmrs-module-formentryapp/commit/debcf32762dc7660f375451d01689b40b518dac7 which was more related to observed stored XSS. I encoded the icon field for the same reason.
The other fields were escaped to fulfill the purpose of the issues for which this pr was made, namely escaping the quote(") character to ensure correct display of those fields.